client/components/auto-direction/index.jsx

```diff
@@ -0,0 +1,181 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import userModule from 'lib/user';
+
+const user = userModule();
+
+/***
+ * Gets text content from react element in case that's a leaf element
+ * @param {React.Element} reactElement react element
+ * @returns {string|null} returns a text content of the react element or null if it's not a leaf element
+ */
+const getContent = ( reactElement ) => {
+	const { props } = reactElement;
+
+	// The child is a text node
+	if ( typeof props.children === 'string' ) {
+		return props.children;
+	}
+
+	// This child has it's content set to external HTML
+	if ( typeof props.dangerouslySetInnerHTML === 'object' ) {
+		// Strip tags because we're only interested in the text, not markup
+		// copied from: http://stackoverflow.com/questions/5002111/javascript-how-to-strip-html-tags-from-string#answer-5002161
+		return props.dangerouslySetInnerHTML.__html
+			? props.dangerouslySetInnerHTML.__html.replace( /<\/?[^>]+(>|$)/g, '' )
+			: '';
+	}
+
+	// This child is some kind of input
+	if ( typeof props.value === 'string' ) {
+		return props.value;
+	}
+
+	// We have no idea how to get this element's content or it's not a leaf component
+	return null;
+};
+
+// Adopted from from: https://github.com/twitter/RTLtextarea/blob/master/src/RTLText.module.js#L25
+const RTLCharRanges = [
+	{
+		name: 'Hebrew',
+		start: 0x590,
+		end: 0x5FF
+	},
+	{
+		name: 'Arabic',
+		start: 0x600,
+		end: 0x6FF
+	},
+	{
+		name: 'Syriac',
+		start: 0x700,
+		end: 0x74F
+	},
+	{
+		name: 'Arabic Supplement',
+		start: 0x750,
+		end: 0x77F
+	},
+	{
+		name: 'Thaana',
+		start: 0x780,
+		end: 0x7BF
+	},
+	{
+		name: 'N\'Ko',
+		start: 0x7C0,
+		end: 0x7FF
+	},
+	{
+		name: 'Samaritan',
+		start: 0x800,
+		end: 0x83F
+	},
+	{
+		name: 'Arabic Extended-A',
+		start: 0x8A0,
+		end: 0x8FF
+	},
+	{
+		name: 'Hebrew presentation forms',
+		start: 0xFB1D,
+		end: 0xFB4F
+	},
+	{
+		name: 'Arabic presentation forms A',
+		start: 0xFB50,
+		end: 0xFDFF
+	},
+	{
+		name: 'Arabic presentation forms B',
+		start: 0xFE70,
+		end: 0xFEFF
+	},
+];
+
+const RTL_THRESHOLD = 0.5;
+const MAX_LENGTH_OF_TEXT_TO_EXAMINE = 1000;
+
+const isRTLChar = ( char ) => {
```

Comment:
Super minor: in general, the whole word seems to be used instead of `char`:

```bash
wp-calypso (add/auto-direction) ☯ spot -i client/ character | wc -l
     262
```